lib/smack/Prelude.cpp

```diff
@@ -253,6 +253,20 @@ FuncDecl* truncBvOp(unsigned width, unsigned width2) {
     makeVars(1, type), type2, body, {makeInlineAttr()});
 }
 
+FuncDecl* bswapBvOp(unsigned width) {
+  std::string type = getBvTypeName(width);
+  std::string var = makeVarName(1);
+  const Expr* body = Expr::bvConcat(Expr::bvExtract(var,width/2,width/2-8),
+                                    Expr::bvExtract(var,width/2+8,width/2));
+  for (unsigned i = 8; i < width/2; i += 8) {
```

Comment:
Does this work for *any* bit-width, for example i37?

Reply:
@keram88 or @shaobo-he - please take a look at this question and maybe address it (or answer it).

Reply:
> Does this work for _any_ bit-width, for example i37?

Not sure I understand this question. These functions are generated only when the bit-width can be divided by 16.

Reply:
Only bit-widths that are a multiple of 16 are allowed by LLVM. Putting other widths in appears to crash Smack when the LLVM code gets first processed. We could still put a check here.

Reply:
I get it. All good. No need for a check here.